src/otaclient_common/cmdhelper.py

```diff
@@ -458,6 +520,54 @@ def mount_ro(
         subprocess_call(cmd, raise_exception=raise_exception)
 
 
+def mount_squashfs(
+    target: StrOrPath, mount_point: StrOrPath, *, raise_exception: bool = True
+) -> None:  # pragma: no cover
+    """Mount the <target> sqiashfs to <mount_point>.
+
+    This is implemented by calling:
+        mount --make-private --make-unbindable -o ro $LOOPDEV -t squashfs <target> <mount_point>
+
+    NOTE: pass args = ["--make-private", "--make-unbindable"] to prevent
+            mount events propagation to/from this mount point.
+
+    Args:
+        target (StrOrPath): target to be mounted.
+        mount_point (StrOrPath): mount point to mount to.
+        raise_exception (bool, optional): raise exception on subprocess call failed.
+            Defaults to True.
+    """
+    # Setup loop device for the squashfs file
+    loop_cmd = ["losetup", "--show", "-f", "-r", str(target)]
+    loop_dev = subprocess_check_output(
+        loop_cmd, raise_exception=raise_exception
+    ).strip()
+
+    if not loop_dev:
+        error_msg = f"Failed to setup loop device for {target}"
+        logger.error(error_msg)
+        if raise_exception:
+            raise RuntimeError(error_msg)
+        return
+
+    # fmt: off
+    cmd = [
+        "mount",
+        "--make-private", "--make-unbindable",
+        "-o", "ro",
+        "-t", "squashfs",
+        loop_dev,
+        str(mount_point),
+    ]
+    # fmt: on
+    try:
+        subprocess_call(cmd, raise_exception=True)
+    except Exception:
+        subprocess_call(["losetup", "-d", loop_dev], raise_exception=False)
+        if raise_exception:
+            raise
+
+
```

Comment:
When `mount` is failed, `losetup -d` is necessary to remove loop devices in addition to `umount`.
Thus, first setup loop device, then mount it. If error is occurred, remove the loope device.